packages/domains-table/src/domains-table-header/columns.ts

```diff
@@ -42,23 +43,23 @@ export const allSitesViewColumns = (
 		isSortable: true,
 		initialSortDirection: 'asc',
 		supportsOrderSwitching: true,
-		sortFunctions: [ getSimpleSortFunctionBy( 'domain' ) ],
+		sortFunctions: [ getSimpleSortFunctionBy( 'owner' ) ],
 	},
 	{
 		name: 'site',
 		label: __( 'Site', __i18n_text_domain__ ),
 		isSortable: true,
 		initialSortDirection: 'asc',
 		supportsOrderSwitching: true,
-		sortFunctions: getSiteSortFunctions(),
+		sortFunctions: [ getSimpleSortFunctionBy( 'blog_name' ) ],
```

Comment:
I think we still need the special-purpose sorting function when sorting by site. The problem with sorting by `blog_name` only is that `blog_name` isn't what's displayed in this column all of the time.

it looks `blog_name` gets set to the domain name when it's a domain-only site. The old sorting method sorted al the domain-only sites to the bottom, and I think this makes sense. If you're specifically wanting to scan your domains by site, it doesn't make sense to have lots of non-sites mixed up with your sites. They should all be moved to the bottom of the table.
In our table we render a "Add site" link in the site column when it's a domain-only site. So it also doesn't look alphabetical.

The other bit of magic sorting we do in the current sorting function is deal with "unnamed" sites. If you go to your site settings and remove the site title, this table will render nothing in the site column. To me this means it should get sorted either right to the top or right to the bottom. However the backend appears to set `blog_name` to the domain name in this case too.

So I think the site column is still a special case and can't be sorted with _just_ `blog_name`

Reply:
>it looks blog_name gets set to the domain name when it's a domain-only site. 

But doesn't this get rendered into the site column via `site.name`, i.e. it is what the user will see?

> If you go to your site settings and remove the site title, this table will render nothing in the site column

Similar to the expiry comment. I think the standard way this gets handled is to show them all first or last. Using `domain` feels confusing if there is no content in the cell or just a dash.